### PR TITLE
Feature/sync queue

### PR DESCRIPTION
**Pull request by jamesmcnamara**
Synchronizes the queue from the player to the backend and creates individual playlist rooms a la appear.in.

Depends on [turntable#18](https://github.com/openrecord/turntable/pull/18) for a local prisma instance. Pull that and follow the instructions in the README to get it started.

Closes turntable#17

### CODE REVIEW

File: src/modules/common/App.jsx [hunk 1]
```diff
@@ -16,7 +16,7 @@ export default function App() {
 				<Switch>
 					<Route exact path={ROUTES.LANDING} component={Landing} />
 					<Route exact path={ROUTES.REGISTER} component={RegisterContainer} />
-					<Route path={ROUTES.PLAYER} component={QueueContainer} />
+					<Route path={ROUTES.PLAYER} component={PlaylistContainer} />
```
[the remark this answers is not in the document]
Comment (rtluu) [reply src/modules/common/App.jsx]: nice this makes a lot of sense

File: src/modules/common/components/Adapt.jsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+		</Component>
+	);
+}
+
+// Extension of React-Adopt (https://github.com/pedronauck/react-adopt)
+// instead of taking a single map of objects, adapt can take multiple
+// maps, and passes all the props collected from the 1->nth maps
+// into each function/component of the n+1th map
+// Allows later components to depend on earlier ones
+export default function adapt(...mappings) {
```
Comment (rtluu) [inline src/modules/common/components/Adapt.jsx]: This is cool

File: src/modules/common/components/WithPlaylistId.jsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+	const {
+		params: {playlist}
+	} = matchPath(pathname, {
+		path: ROUTES.PLAYER,
+		exact: true
+	});
+	return children(playlist);
+}
+
+export default withRouter(WithPlaylistId);
```
[the remark this answers is not in the document]
Comment (rtluu) [reply src/modules/common/components/WithPlaylistId.jsx]: Nice this is super useful

File: src/modules/common/fragments/TrackFragments.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+				id
+				thumbnail
+				title
+				description
+				url
+				source
+			}
+		}
+	`
+};
```
[the remark this answers is not in the document]
Comment (rtluu) [reply src/modules/common/fragments/TrackFragments.js]: Got it, kind of like a chained event

File: src/modules/queue/components/Queue.jsx [hunk 5]
```diff
@@ -1,48 +1,24 @@
...
-			</QueueList>
-		</StyledQueue>
+		<QueueList>
+			{tracks.map(track => (
+				<Track
+					{...track.info}
+					key={track.id}
+					onClick={() => updatePlaying(track)}
+					playing={currentlyPlayingId === track.id}
+					highRes
```
[the remark this answers is not in the document]
Comment (rtluu) [reply src/modules/queue/components/Queue.jsx]: Yea this is an important change and I like the name 'Track' 😃

File: src/modules/routes/routes.js [hunk 6]
```diff
@@ -2,5 +2,5 @@ export const ROUTES = {
 	LANDING: '/',
 	REGISTER: '/register',
 	PROFILE: '/profile',
-	PLAYER: '/player'
+	PLAYER: '/:playlist'
```
[the remark this answers is not in the document]
Comment (rtluu) [reply src/modules/routes/routes.js]: 🙌🏼

File: src/modules/search/components/SearchContainer.jsx [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+					title
+					url
+					thumbnail
+					description
+					source
+				}
+			}
+		}
+	}
+`;
```
[the remark this answers is not in the document]
Comment (rtluu) [reply src/modules/search/components/SearchContainer.jsx]: So just to solidify my understanding, for adding a song we query GraphQL to see if that url exists, if it does, it draws the information and updates the playlist, if not it creates the track and updates the playlist. 

That way, each time the same track is interacted with across OpenRecord the cumulative data is shared. That's awesome 🔥

File: src/modules/search/components/Track.jsx [hunk 8]
```diff
@@ -1,22 +1,12 @@
...
-		description,
-		thumbnails: {
-			default: {url}
-		}
-	}
-}) {
+export default function Track({onClick, playing, search, thumbnail, title}) {
 	return (
-		<StyledResult onClick={onClick}>
+		<StyledResult onClick={onClick} className={classname({playing})}>
```
[the remark this answers is not in the document]
Comment (rtluu) [reply src/modules/search/components/Track.jsx]: 😉love it haha

File: src/modules/search/components/SearchContainer.jsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (jamesmcnamara) [reply src/modules/search/components/SearchContainer.jsx]: Yep!

If we want to be nitpicky, we "use GraphQL to query our Prisma server to see if that url exists"

**Pull request merged**